pkg/controller/clusterpool/collections.go

```diff
@@ -796,6 +796,13 @@ func (cdcs *cdcCollection) Unassign(c client.Client, cdc *hivev1.ClusterDeployme
 
 	delete(cdcs.reserved, cdc.Name)
 
+	// remove duplicates
```

Comment:
How does duplication happen? Wouldn't it mean the CDC was already unassigned?

It looks like it might come from [here](https://github.com/openshift/hive/pull/1855/files#diff-bbf4d3cb24b8cac87951b84ca973747902c00d4a9d662648c8e760a37569581dR999) -- the condition surrounding this dictates that the CDC is available, so why wouldn't it have been in the unassigned list in the first place? Seems like that's where we should attack this situation, rather than culling duplicates after the fact.

Is this related to the rest of the PR? If not, I would like to see it split out.

Reply:
The duplication comes from [here](https://github.com/openshift/hive/blob/master/pkg/controller/clusterpool/collections.go#L992) (I think we are pointing at the same code). Its unrelated to this PR, just another issue I have stumbled upon. I will create another PR.